static/js/hoc/withPostList_test.js

```diff
@@ -0,0 +1,218 @@
+  render() {
+    const { renderPosts } = this.props
+    return <div>{renderPosts()}</div>
+  }
+}
+
+const WrappedPage = withPostList(Page)
+
+describe("withPostList", () => {
+  let helper, render, postList, loadPosts
+
+  beforeEach(() => {
+    helper = new IntegrationTestHelper()
+    loadPosts = helper.sandbox.stub()
+    postList = makeChannelPostList()
+    render = helper.configureHOCRenderer(
+      WrappedPage,
+      Page,
+      {},
+      {
+        posts:     postList,
+        loadPosts: loadPosts,
+        location:  { search: {} }
+      }
+    )
+  })
+
+  afterEach(() => {
+    helper.cleanup()
+  })
+
+  describe("pagination", () => {
+    [{ sort: "new" }, null].forEach(pagination => {
+      it(`has PostList and ${
+        pagination ? "" : "no "
+      }InfiniteScroll if there is ${
+        pagination ? "" : "no "
+      }pagination`, async () => {
+        const { inner } = await render(
+          {},
+          {
+            pagination
+          }
+        )
+        assert.equal(inner.find("InfiniteScroll").length, pagination ? 1 : 0)
+        assert.equal(inner.find("PostList").length, 1)
+      })
+    })
+    ;["after", null].forEach(afterUrl => {
+      it(`sets hasMore=${String(!!afterUrl)} if it ${
+        afterUrl ? "has" : "doesn't have"
+      } an after url`, async () => {
+        const { inner } = await render(
+          {},
+          {
+            pagination: {
+              sort:  "new",
+              after: afterUrl
+            }
+          }
+        )
+
+        assert.equal(inner.find("InfiniteScroll").props().hasMore, !!afterUrl)
+      })
+    })
+
+    describe("loadMore", () => {
+      let pagination
+
+      beforeEach(() => {
+        pagination = makePagination()
+      })
+
+      it("doesn't do anything if canLoadMore is not set", async () => {
+        const { inner } = await render(
+          {},
+          {
+            canLoadMore: false,
+            pagination:  pagination
+          }
+        )
+        const props = inner.find("InfiniteScroll").props()
+        await props.loadMore()
+        sinon.assert.notCalled(loadPosts)
+      })
+      ;[true, false].forEach(hasParams => {
+        it(`${
+          hasParams ? "uses" : "doesn't use"
+        } query parameters from the URL for the pagination`, async () => {
+          const { inner } = await render(
+            {},
+            {
+              canLoadMore: true,
+              pagination:  pagination,
+              location:    {
+                search: hasParams ? "?sort=some_sort&after=different_after" : ""
+              }
+            }
+          )
+          const props = inner.find("InfiniteScroll").props()
+          await props.loadMore()
+          assert.equal(loadPosts.callCount, 1)
+          sinon.assert.calledWith(loadPosts, {
+            after: pagination.after,
+            count: pagination.after_count,
+            ...(hasParams ? { sort: "some_sort" } : {})
```

Comment:
Can you parametrize the `location.search` value and the expected expected `loadPosts` params?